app/jobs/repo_setup_job.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+require "concerns/repo_setup.rb"
+
+class RepoSetupJob < ApplicationJob
+  queue_as :repo_setup
+
+  def perform(assignment_repo_type, assignment_repo_id)
+    @assignment_repo = get_assignment_repo assignment_repo_type, assignment_repo_id
```

Comment:
Though if we pass the object instead of the type and id, we won't need to do this at all